Issue #289: Document the most common use case (#481)
sentrysoftware/metricshub: metricshub-doc/src/site/markdown/usecases/grafana-service.md

# Monitoring a Grafana Service

## Overview

If the connectors available with **MetricsHub** do not collect the metrics you need, you can configure one or several monitors to obtain this data from your resource and specify its corresponding attributes and metrics in **MetricsHub**.

In the example below, we created a monitor that collects data from the Grafana health API and map the output to **MetricsHub**'s attributes and metrics. 

## Procedure

To collect data from the Grafana health API:

1. In the `config/metricshub.yaml` file, we first created the `grafana-service` service group, specified its attributes and the protocol to use to connect to this service group:

    ```yaml
    service-group:  
      grafana-service:
        attributes:
          service.name: Grafana
          host.name: hws-demo.sentrysoftware.com
        protocols:
          http:
            https: true
            port: 443
    ```

2. We then specified the job to be performed. In our example, we opted for `simple` for straightforward monitoring tasks:  

    ```yaml
      monitors:
        grafana:
          simple: 
    ```

> 3. We specified the  Grafana health API data source as follows:

Reviewer: ```suggestion
3. We specify the  Grafana health API as a data source as follows:
```

    ```yaml
              sources:
                grafanaHealth:
                  type: http
                  path: /api/health
                  method: get
                  header: "Accept: application/json"
                  computes:
                  - type: json2Csv
                    entryKey: /
                    properties: commit;database;version
                    separator: ;
                  - type: translate
                    column: 3
                    translationTable:
                      ok: 1
                      default: 0
    ```
4. Finally, we specified how the collected metrics are mapped to **MetricsHub**'s monitoring model:

    ```yaml
    service-group:  
      grafana-service:
        attributes:
          service.name: Grafana
          host.name: hws-demo.sentrysoftware.com
        protocols:
          http:
            https: true
            port: 443
        monitors:
          grafana:
            simple: 
              sources:

>                 grafanaHealth:

Reviewer: ```suggestion
                # HTTP GET /api/health which returns a JSON
                grafanaHealthApi:
```

                  type: http
                  path: /api/health
                  method: get
                  header: "Accept: application/json"
                  computes:

>                   - type: json2Csv

Reviewer: ```suggestion
                  # Convert the JSON document into a simple CSV
                  - type: json2Csv
```

                    entryKey: /
                    properties: commit;database;version
                    separator: ;

>                   - type: translate

Reviewer: ```suggestion
                  # Translate the value of "database" in the 2nd column:
                  # - "ok" --> 1
                  # - anything else --> 0
                  - type: translate
```

                    column: 3
                    translationTable:
                      ok: 1
                      default: 0

>               mapping:

Reviewer: Remove all YAML code above `mapping:`

>                 source: ${esc.d}{source::grafanaHealth}

Reviewer: ```suggestion
                source: ${esc.d}{source::grafanaHealthApi}
```

                attributes:
                  id: $2
                  service.instance.id: $2
                  service.version: $4
                metrics:

>                   grafana.db.state: $3

Reviewer: ```suggestion
                  grafana.db.status{state="ok"}: $3
```

    ```

> We came up to this version of the `config/metricshub.yaml` configuration file:

Reviewer: Here is the complete YAML configuration to be added to `config/metricshub.yaml` to monitor the health of our Grafana service:

> ```yaml

Reviewer: Update the YAML as per previous comments
